### PR TITLE
[Handshake] Expose bufferRegion for other users

### PR DESCRIPTION
**Pull request by Dinistro**
This commit exposes a new `bufferRegion` function that is independent of `handshake.func`. Additionally, it cleans up some of the code in the buffer insertion.

### CODE REVIEW

File: test/Dialect/Handshake/compose-buffers.mlir [hunk 1]
```diff
@@ -24,3 +24,9 @@ handshake.func @foo(%arg0 : i32, %ctrl : none) -> (i32, none) {
   %2:2 = fork [2] %1 : i32
   return %2#1, %ctrl : i32, none
 }
+
+// -----
+
+// CHECK-LABEL:   handshake.func @external(
+// CHECK-SAME:      i32, none, ...) -> none
+handshake.func @external(%arg0: i32, %ctrl: none, ...) -> none
```
Comment (mortbopet) [inline test/Dialect/Handshake/compose-buffers.mlir:32]: Why is this here?

Comment (Dinistro) [reply test/Dialect/Handshake/compose-buffers.mlir:32]: Oh, I thought that this is the file to test buffer insertion. It seems that there is in fact none ;/
This case wasn't tested before, so while refactoring, I added a check for external functions and added a test.

I should have maybe split this into a separate PR...
